### PR TITLE
EOS-24922: Confstore: consul should be able to connect to consul cluster outside localhost

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

# Problem Statement
- Consul was not able to connect to dc present on other hosts. It expects that host should be part of consul cluster

# Design
-  Added host and port information while creating the consul object

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: No
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [x] Changes done to WIKI / Confluence page
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/316342315/Conf+Store

### CODE REVIEW

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_store_collection.py [hunk 1]
```diff
@@ -465,7 +465,14 @@ class ConsulKVStore(KvStore):
 
     def __init__(self, store_loc, store_path, delim='>'):
         KvStore.__init__(self, store_loc, store_path, delim)
-        self.c = Consul()
+        if store_loc:
+            if ":" in store_loc:
+                host, port = store_loc.split(":")
```
Comment (vastradparayya) [inline py-utils/src/utils/kv_store/kv_store_collection.py:470]: Should be single quotes

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by suryakumar1024 — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_store_collection.py (hunk 1 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_store_collection.py]: Done

**Review by vastradparayya — APPROVED** [review]

**Comment by rahul-tripathi-git** [comment]
Custom build: http://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/centos-7.9/job/custom-ci/2105/
manual deployment: http://eos-jenkins.mero.colo.seagate.com/job/Cortx-Deployment/job/VM-Deployment-1Node-CentOS-7.9/168/

**Pull request merged**